File: PokerFrontendReact/src/services/apiService.js
import axios from 'axios';

const API_BASE_URL = '/api/GameControllerAPI';

const apiService = {
    registerPlayer: async (name, chips) => {
        // Note: The backend uses query parameters for POST requests in some cases based on the controller code.
        // Ensure we match the backend expectation (FromQuery vs FromBody).
        // Based on GameControllerAPI.cs: [HttpPost("registerPlayer")] public ... ([FromQuery] string playerName, [FromQuery] int chipStack)
        return axios.post(`${API_BASE_URL}/registerPlayer?playerName=${name}&chipStack=${chips}`);
    },

    joinSeat: async (playerName, seatIndex) => {
        // [HttpPost("joinSeat")] ... ([FromQuery] string playerName, [FromQuery] int seatIndex)
        return axios.post(`${API_BASE_URL}/joinSeat?playerName=${playerName}&seatIndex=${seatIndex}`);
    },

    startRound: async () => {
        return axios.post(`${API_BASE_URL}/startRound`);
    },

    bet: async (playerName, amount) => {
        return axios.post(`${API_BASE_URL}/bet?name=${playerName}&amount=${amount}`);
    },

    call: async (playerName) => {
        return axios.post(`${API_BASE_URL}/call?name=${playerName}`);
    },

    check: async (playerName) => {
        return axios.post(`${API_BASE_URL}/check?name=${playerName}`);
    },

    fold: async (playerName) => {
        return axios.post(`${API_BASE_URL}/fold?name=${playerName}`);
    },

    raise: async (playerName, amount) => {
        return axios.post(`${API_BASE_URL}/raise?name=${playerName}&amount=${amount}`);
    },

    allIn: async (playerName) => {
        return axios.post(`${API_BASE_URL}/allin?name=${playerName}`);
    },

    getGameState: async () => {
        return axios.get(`${API_BASE_URL}/state`);
    },

    removePlayer: async (name) => {
        return axios.post(`${API_BASE_URL}/removePlayer?name=${name}`);
    },

    addChips: async (playerName, amount) => {
        return axios.post(`${API_BASE_URL}/addchips`, {
            PlayerName: playerName,
            Amount: amount
        });
    },

    nextPhase: async () => {
        return axios.post(`${API_BASE_URL}/nextPhase`);
    },

    showdown: async () => {
        return axios.post(`${API_BASE_URL}/showdown`);
    }
};

export default apiService;

File: Sandbox/PokerUIClient/Services/ApiService.cs
using System.Net.Http;
using System.Net.Http.Json;
using PokerUIClient.Models;

namespace PokerUIClient.Services
{
    public class ApiService
    {
        private readonly HttpClient _http;
        private const string BaseUrl = "http://localhost:5175/api/GameControllerAPI/";

        public ApiService(HttpClient http)
        {
            _http = http;
        }

        // =========================
        // Game State
        // =========================
        public async Task<GameStateDTO?> GetStateAsync()
        {
            try
            {
                var resp = await _http.GetAsync($"{BaseUrl}state");
                if (!resp.IsSuccessStatusCode) return null;
                return await resp.Content.ReadFromJsonAsync<GameStateDTO>();
            }
            catch
            {
                return null;
            }
        }

        // =========================
        // Player Management
        // =========================
        public async Task<bool> AddPlayerAsync(string name, int chips = 1000, int seatIndex = -1)
        {
            try
            {
                var url = $"{BaseUrl}addPlayer?name={Uri.EscapeDataString(name)}&chips={chips}&seatIndex={seatIndex}";
                var resp = await _http.PostAsync(url, null);
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }

        public async Task<bool> RegisterPlayerAsync(string name, int chips)
        {
            try
            {
                var url = $"{BaseUrl}registerPlayer?playerName={Uri.EscapeDataString(name)}&chipStack={chips}";
                var resp = await _http.PostAsync(url, null);
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }

        public async Task<bool> RemovePlayerAsync(string name)
        {
            try
            {
                var url = $"{BaseUrl}removePlayer?name={Uri.EscapeDataString(name)}";
                var resp = await _http.PostAsync(url, null);
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }

        public async Task<bool> JoinSeatAsync(string name, int seatIndex)
        {
            try
            {
                var url = $"{BaseUrl}joinSeat?playerName={Uri.EscapeDataString(name)}&seatIndex={seatIndex}";
                var resp = await _http.PostAsync(url, null);
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }

        public async Task<bool> AddChipsAsync(string name, int amount)
        {
            try
            {
                var resp = await _http.PostAsJsonAsync($"{BaseUrl}addchips", new { PlayerName = name, Amount = amount });
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }

        // =========================
        // Round Management
        // =========================
        public async Task<bool> StartRoundAsync()
        {
            try
            {
                var resp = await _http.PostAsync($"{BaseUrl}startRound", null);
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }

        public async Task<bool> NextPhaseAsync()
        {
            try
            {
                var resp = await _http.PostAsync($"{BaseUrl}nextPhase", null);
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }

        public async Task<ShowdownDTO?> ShowdownAsync()
        {
            try
            {
                var resp = await _http.PostAsync($"{BaseUrl}showdown", null);
                if (!resp.IsSuccessStatusCode) return null;
                return await resp.Content.ReadFromJsonAsync<ShowdownDTO>();
            }
            catch
            {
                return null;
            }
        }

        // =========================
        // Player Actions
        // =========================
        public async Task<bool> BetAsync(string name, int amount)
        {
            try
            {
                var url = $"{BaseUrl}bet?name={Uri.EscapeDataString(name)}&amount={amount}";
                var resp = await _http.PostAsync(url, null);
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }

        public async Task<bool> RaiseAsync(string name, int amount)
        {
            try
            {
                var url = $"{BaseUrl}raise?name={Uri.EscapeDataString(name)}&amount={amount}";
                var resp = await _http.PostAsync(url, null);
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }

        public async Task<bool> CallAsync(string name)
        {
            try
            {
                var url = $"{BaseUrl}call?name={Uri.EscapeDataString(name)}";
                var resp = await _http.PostAsync(url, null);
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }

        public async Task<bool> CheckAsync(string name)
        {
            try
            {
                var url = $"{BaseUrl}check?name={Uri.EscapeDataString(name)}";
                var resp = await _http.PostAsync(url, null);
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }

        public async Task<bool> FoldAsync(string name)
        {
            try
            {
                var url = $"{BaseUrl}fold?name={Uri.EscapeDataString(name)}";
                var resp = await _http.PostAsync(url, null);
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }

        public async Task<bool> AllInAsync(string name)
        {
            try
            {
                var url = $"{BaseUrl}allin?name={Uri.EscapeDataString(name)}";
                var resp = await _http.PostAsync(url, null);
                return resp.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }
    }
}
